docs/dev/benchmarking.md

```diff
@@ -0,0 +1,138 @@
+# Benchmarking
+
+This document gives an overview of the benchmarking infrastructure of Concrete.
+
+## Concrete Python
+
+Concrete Python uses [progress-tracker-python](https://github.com/zama-ai/progress-tracker-python) to do benchmarks. Please refer to its README to learn how it works.
+
+### How to run all benchmarks?
+
+Use a makefile target:
+
+```shell
+make benchmark
+```
+
+Note that this command removes the previous benchmark results before doing the benchmark.
+
+### How to run a single benchmark?
+
+Since the full benchmark suite takes a long time to run, it's not recommended for development. Instead, use the following command to run just a single benchmark.
+
+```shell
+TARGET=foo make benchmark-target
+```
+
+This command would only run the benchmarks defined in `benchmarks/foo.py`. It also retains the previous runs, so it can be run back to back to collect data from multiple benchmarks.
+
+### What are benchmarked?
+
+Every python file inside `benchmarks` directory is a benchmark script:
+
+### primitive.py
+
+Purpose: benchmarking simple operations (e.g., tlu, matmul, ...)
+
+Parameters:
+- operation
+- inputset (e.g., different bit-widths, different shapes)
+- configuration (e.g., different strategies)
+
+Collected Metrics:
+- Compilation Time (ms)
+- Complexity
+- Key Generation Time (ms)
+- Evaluation Key Size (MB)
+- Input Ciphertext Size (MB)
+- Output Ciphertext Size (MB)
+- Encryption Time (ms)
+- Evaluation Time (ms)
+- Decryption Time (ms)
+- Accuracy
+
+### static_kvdb.py
```

Comment:
not sure we want to keep updated the list of *.py which are not primitive.py, do we? ie, if we add a pir.py, we shouldn't add it here, should we? instead, I would mention there is primitive.py and that there are the other .py's for benchmarking the different usecases.

Reply:
I'd say it's better to put here, just to give an overview without looking to every use case manually.

Reply:
but you mean, with every new tests, people will have to update this page? for sure they'll never think about it

Reply:
We should pay attention to this in the PRs. We might miss some initially, but we can fix those as we found them. It'll be better in any case.

Reply:
one more tie for @BourgerieQuentin ! @BourgerieQuentin , what do you think here, please?

Reply:
Yes I agree with @bcm-at-zama , I think this should be just a documenation of the benchmark itself.

Reply:
Okay, I'll remove this section then.